src/ripple/nodestore/ShardValidation.md

```diff
@@ -0,0 +1,130 @@
+# Downloaded Shard Validation
+
+## Overview
+
+In order to validate shards that have been downloaded from file servers (as opposed to shards acquired from peers), the application must confirm the validity of the downloaded shard's last ledger. The following sections describe this confirmation process in greater detail.
+
+## Execution Concept
+
+### Flag Ledger
+
+Since the number of ledgers contained in each shard is always a multiple of 256, a shard's last ledger is always a flag ledger. Conveniently, the application provides a mechanism for retrieving the hash for a given flag ledger:
+
+```C++
+boost::optional<uint256>
+hashOfSeq (ReadView const& ledger,
```

Comment:
Because the skip list is guaranteed to store all flag ledgers with a sequence number earlier than the most recently validated ledger. So we can count on being able to retrieve the flag ledger.

Reply:
Sorry, I was not clear.  If you look at the signature of `hashOfSeq()`, there's nothing in there about flag ledgers.  So referring to that particular function does nothing to reinforce the importance of using flag ledgers.  Your point would be better reinforced by mentioning the existence of the skip list in the ledger, not by the `hashOfSeq()` function.

Reply:
I added a line about the skip list and mentioned that we access it via `walkHashBySeq`.